services/graph/pkg/service/v0/base.go

```diff
@@ -31,6 +31,11 @@ import (
 	"github.com/owncloud/ocis/v2/services/graph/pkg/unifiedrole"
 )
 
+// BaseGraphProvider is the interface that wraps shared methods between the different graph providers
+type BaseGraphProvider interface {
+	CS3ReceivedSharesToDriveItems(ctx context.Context, receivedShares []*collaboration.ReceivedShare) ([]libregraph.DriveItem, error)
+}
```

Comment:
Hm, do we really need this? We just using it for mocking in a single test case. (Basically because of the changed return of `MountShare` method. Again, not strictly against this, but I'd like to understand the reasons.

Reply:
i thought its easier to pass around one base service than pass down identityCache and logger, it also guarantees that the call always uses a rotated gatewayClient.

i like the idea of having a low-level service which contains the building blocks all in one.